include/swift/Syntax/SyntaxClassifier.h.gyb

```diff
@@ -0,0 +1,114 @@
+%{
+  # -*- mode: C++ -*-
+  from gyb_syntax_support import *
+  NODE_MAP = create_node_map()
+  # Ignore the following admonition; it applies to the resulting .h file only
+}%
+//// Automatically Generated From SyntaxClassifier.h.gyb.
+//// Do Not Edit Directly!
+//===----------- SyntaxClassifier.h - SyntaxClassifier definitions --------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_SYNTAX_CLASSIFIER_H
+#define SWIFT_SYNTAX_CLASSIFIER_H
+
+#include "swift/AST/Identifier.h"
```

Comment:
This is layer violation. Could you move `bool isEditorPlaceholder(StringRef name)` to `lib/Basic/EditorPlaceholder.cpp`?
Also, if you don't use it in `.h`, move this to `.cpp`.

Reply:
Is there any documentation of what the layering is supposed to be?

Good catch. Forgot to move it when I moved stuff from the `.h` file to `.cpp`.

Reply:
> Is there any documentation of what the layering is supposed to be?

I don't think so :(